From d20d46a62960099164da5ef5bb7b0b93127cc6c5 Mon Sep 17 00:00:00 2001
From: Aleksander Morgado <aleksander@aleksander.es>
Date: Thu, 23 Nov 2017 14:10:59 +0100
Subject: [PATCH 1/3] iface-modem: remove invalid modem reference count in
 update_lock_info()

The modem object is being explicitly referenced and stored in the
Context, but then never unref-ed, completely leaking a modem reference
forever.

Fixes: 4df545884733bbc5a851ab86e0983dec057d5482
---
 src/mm-iface-modem.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/mm-iface-modem.c b/src/mm-iface-modem.c
index 92cd00a1..afb3be32 100644
--- a/src/mm-iface-modem.c
+++ b/src/mm-iface-modem.c
@@ -2986,9 +2986,7 @@ typedef enum {
 } UpdateLockInfoContextStep;
 
 typedef struct {
-    MMIfaceModem *self;
     UpdateLockInfoContextStep step;
-    GSimpleAsyncResult *result;
     MmGdbusModem *skeleton;
     MMModemLock lock;
     GError *saved_error;
@@ -3230,7 +3228,6 @@ mm_iface_modem_update_lock_info (MMIfaceModem *self,
     GTask *task;
 
     ctx = g_slice_new0 (UpdateLockInfoContext);
-    ctx->self = g_object_ref (self);
     g_object_get (self,
                   MM_IFACE_MODEM_DBUS_SKELETON, &ctx->skeleton,
                   NULL);
-- 
2.15.0

